zerver/views/users.py

```diff
@@ -554,3 +554,28 @@ def get_subscription_backend(request: HttpRequest, user_profile: UserProfile,
     subscription_status = {'is_subscribed': subscribed_to_stream(target_user, stream_id)}
 
     return json_success(subscription_status)
+
+@has_request_variables
+def get_user_by_email(request: HttpRequest, user_profile: UserProfile, email: str,
+                      include_custom_profile_fields: bool=REQ(validator=check_bool,
+                                                              default=False),
+                      client_gravatar: bool=REQ(validator=check_bool, default=False),
+                      ) -> HttpResponse:
+    realm = user_profile.realm
+    if realm.email_address_visibility != Realm.EMAIL_ADDRESS_VISIBILITY_EVERYONE:
+        # If email addresses are only available to administrators,
+        # clients cannot compute gravatars, so we force-set it to false.
+        client_gravatar = False
+    target_user = None
+    if email is not None:
+        target_user = get_user_by_delivery_email(email, realm)
+
+    members = get_raw_user_data(realm, user_profile, target_user=target_user,
+                                client_gravatar=client_gravatar,
+                                user_avatar_url_field_optional=False,
+                                include_custom_profile_fields=include_custom_profile_fields)
+
+    if target_user is not None:
+        data: Dict[str, Any] = {"user": members[target_user.id]}
+
+    return json_success(data)
```

Comment:
Could we write this function as just basically
```
     target_user = get_user(email, realm)
    return get_members_backend(request, user_profile, user_id=target_user.id)
```

to avoid duplicating code?  We already support a specific target user in `get_members_backend`, so it doesn't really make sense to copy its implementation just to allow doing that.

Reply:
Sure